targets/mgyamlfmt/targets.go

```diff
@@ -0,0 +1,54 @@
+package mgyamlfmt
+
+import (
+	"bytes"
+	"io/fs"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"go.einride.tech/mage-tools/mglog"
+	"go.einride.tech/mage-tools/mgpath"
+	"gopkg.in/yaml.v3"
+)
+
+func FormatYaml() error {
+	logger := mglog.Logger("format-yaml")
+	logger.Info("formatting yaml files...")
+	return filepath.WalkDir(mgpath.FromGitRoot("."), func(path string, d fs.DirEntry, err error) error {
+		if filepath.Ext(path) == ".yml" || filepath.Ext(path) == ".yaml" {
+			if err := formatFile(path); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func formatFile(path string) error {
+	node := yaml.Node{}
+	yamlFile, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	// Insert temporary comment to preserve empty lines.
+	yamlFile = bytes.ReplaceAll(yamlFile, []byte("\n\n"), []byte("\n#preserveEmptyLine\n"))
+	if err := yaml.Unmarshal(yamlFile, &node); err != nil {
+		return err
+	}
+	var b bytes.Buffer
+	encoder := yaml.NewEncoder(&b)
+	encoder.SetIndent(2)
+	if err := encoder.Encode(&node); err != nil {
+		return err
+	}
+	// Remove temporary comment.
+	indentPreserveComment := regexp.MustCompile("\n\\s+#preserveEmptyLine\n")
+	out := b.String()
```

Comment:
@viktorvoltaire Minor thought: The conversion to string and back to bytes here should be avoidable. (Converting back and forth allocates memory for a copy).

Reply:
Ah, will have a look and see if I can fix that 👍

Reply:
@odsod actually I already fixed that. Was annoyed with it 😊 https://github.com/einride/mage-tools/commit/ee6ea005bfd2721fd1165f58320a72b09ec3fe0c#diff-bd1f7610004a598ff323f80c6917c7ede5455328026fc0397bc78b7554a93d74R38

Reply:
Sweet